### ISSUE TITLE
Memory leak problem 

### ISSUE THREAD

**Issue by xiejiang2014** [SuRGeoNix/Flyleaf#40]
I tried to create a player in the background code. I found that every time I created it, the memory could not be released normally.
How can I improve it?
Thank you.

using FlyleafLib 3.1.0 from nuget

MainWindow.xaml:
```
<Window
    x:Class="FlyleafLib.Test.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    Title="MainWindow"
    Width="800"
    Height="450"
    mc:Ignorable="d">
    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto" />
            <RowDefinition Height="*" />
        </Grid.RowDefinitions>

        <Button Click="ButtonNew_OnClick">
            Create New Instance
            (1. each click leak some memory,
            2. click in quick succession the program crashes )
        </Button>

        <ContentControl
            x:Name="ContentControl"
            Grid.Row="1"
            HorizontalAlignment="Stretch"
            VerticalAlignment="Stretch" />
    </Grid>
</Window>
```

MainWindow.xaml.cs
```
using System;
using System.IO;
using System.Threading.Tasks;
using System.Windows;
using FlyleafLib.Controls.WPF;
using FlyleafLib.MediaPlayer;

namespace FlyleafLib.Test
{
    public partial class MainWindow
    {
        public Player Player { get; set; }

        public MainWindow()
        {
            InitializeComponent();

            Master.RegisterFFmpeg(":2");
        }

        private FlyleafLib.Controls.WPF.VideoView _videoView;

        private FlyleafLib.MediaPlayer.Player _player;

        private FlyleafLib.Controls.WPF.Flyleaf _flyleaf;

        private async void ButtonNew_OnClick(object sender, RoutedEventArgs e)
        {
            _player?.Stop();
            await Task.Run(() => _player?.Dispose());

            _videoView = new VideoView();
            _videoView.UpdateDefaultStyle();
            _videoView.ApplyTemplate();

            _flyleaf           = new Flyleaf();
            _videoView.Content = _flyleaf;

            Config config = new Config();
            config.demuxer.VideoFormatOpt.Add("probesize",       (50 * (long) 1024 * 1024).ToString());
            config.demuxer.VideoFormatOpt.Add("analyzeduration", (10 * (long) 1000 * 1000).ToString());
            _player = new Player(config);

            _videoView.BeginInit();

            _player.Control   = _videoView.FlyleafWF;
            _videoView.Player = _player;
            _flyleaf.Player   = _player;

            ContentControl.Content = _videoView;

            _player.Open(@"VIDEO FILE");

            _player.Play();
        }
    }
}
```

**Comment by SuRGeoNix** [comment]
Hi xiejiang2014, 

Just did a commit related to this one. I fixed mainly the rendering live objects that were causing the main memory leaks. There are still some minor memory leaks related with the audio.

The other issue is that by using the Flyleaf WPF control for that causes more memory leaks. Ideally, you should use a naked control at least for now until I will review also the WPF Control's proper disposing with the VideoView. Have a look at the latest commit (check the sample 'Issue40') and let me know what you think.

Thanks for submitting this.

**Comment by ssummer1337** [comment]
These latest changes introduce a new memory leak:
https://github.com/SuRGeoNix/Flyleaf/blob/05c307b9f5898d077e5aee2a2a31c1b0ae7349ac/FlyleafLib.Controls.WPF/Flyleaf.xaml.cs#L179

The variable Player gets nulled before the Task is being started, therefore I doesn't get disposed.

**Comment by SuRGeoNix** [comment]
Should be now fixed on 3.1.9. If not let me know.
